File: examples/delivery/http/java/HttpAgent.java
package ai.axme.examples.delivery;

import ai.axme.examples.SseHelper;
import dev.axme.sdk.*;
import com.sun.net.httpserver.*;

import java.io.*;
import java.net.InetSocketAddress;
import java.util.*;

/** Order Processing Agent — HTTP delivery binding (Java). */
public class HttpAgent {
    static final String BASE = SseHelper.env("AXME_BASE_URL", "https://api.cloud.axme.ai");
    static final String KEY = System.getenv("AXME_API_KEY");
    static final String ADDR = SseHelper.env("AXME_AGENT_ADDRESS", "http-order-processor");
    static final int PORT = Integer.parseInt(SseHelper.env("PORT", "8080"));
    static final Set<String> CURRENCIES = Set.of("USD", "EUR", "GBP", "JPY");

    static AxmeClient client;

    public static void main(String[] args) throws Exception {
        if (KEY == null || KEY.isEmpty()) { System.err.println("AXME_API_KEY is required"); System.exit(1); }
        client = new AxmeClient(new AxmeClientConfig(BASE, KEY));

        HttpServer server = HttpServer.create(new InetSocketAddress(PORT), 0);
        server.createContext("/health", ex -> { ex.sendResponseHeaders(200, 15); ex.getResponseBody().write("{\"status\":\"ok\"}".getBytes()); ex.close(); });
        server.createContext("/intent", ex -> {
            if (!"POST".equals(ex.getRequestMethod())) { ex.sendResponseHeaders(405, -1); ex.close(); return; }
            byte[] body = ex.getRequestBody().readAllBytes();
            ex.sendResponseHeaders(200, 10); ex.getResponseBody().write("{\"ack\":true}".getBytes()); ex.close();
            Map<String, Object> envelope = SseHelper.parseJson(new String(body));
            String id = SseHelper.str(envelope, "intent_id");
            if (!id.isEmpty()) new Thread(() -> handleAsync(id)).start();
        });
        server.start();
        System.out.printf("http-order-processor starting  port=%d%n", PORT);
    }

    static void handleAsync(String id) {
        try {
            Map<String, Object> intent = client.getIntent(id, RequestOptions.none());
            @SuppressWarnings("unchecked")
            Map<String, Object> p = (Map<String, Object>) intent.getOrDefault("payload", Map.of());
            String eventType = SseHelper.str(p, "event_type");
            String orderId = SseHelper.str(p, "order_id");
            double amount = p.get("amount") instanceof Number n ? n.doubleValue() : 0;
            String currency = SseHelper.str(p, "currency");

            Map<String, Object> res = new HashMap<>();
            if (!"order.placed".equals(eventType)) { res.put("action", "fail"); res.put("reason", "unexpected event_type"); }
            else if (orderId.isEmpty()) { res.put("action", "fail"); res.put("reason", "order_id is required"); }
            else if (!CURRENCIES.contains(currency)) { res.put("action", "fail"); res.put("reason", "unsupported currency"); }
            else {
                res.put("action", "complete"); res.put("order_id", orderId);
                res.put("tracking_id", "TRK-" + orderId.toUpperCase() + "-" + String.format("%05d", Math.abs(orderId.hashCode()) % 100000));
                res.put("status", "accepted"); res.put("currency", currency); res.put("amount", amount);
            }
            client.resumeIntent(id, res, SseHelper.withOwner(ADDR));
            System.out.printf("resumed %s action=%s%n", id, res.get("action"));
        } catch (Exception e) { System.err.println("handle error: " + e.getMessage()); }
    }
}

File: examples/delivery/http/dotnet/HttpAgent.cs
using System.Net;
using System.Text.Json;
using System.Text.Json.Nodes;
using Axme.Sdk;
using AxmeExamples;

namespace AxmeExamples.Delivery;

public class HttpAgent
{
    public static async Task Main(string[] args)
    {
        var baseUrl = SseHelper.Env("AXME_BASE_URL", "https://api.cloud.axme.ai");
        var apiKey = SseHelper.Env("AXME_API_KEY");
        var addr = SseHelper.Env("AXME_AGENT_ADDRESS", "http-order-processor");
        var port = int.Parse(SseHelper.Env("PORT", "8080"));
        if (string.IsNullOrEmpty(apiKey)) { Console.Error.WriteLine("AXME_API_KEY is required"); return; }

        var client = new AxmeClient(new AxmeClientConfig { ApiKey = apiKey, BaseUrl = baseUrl });
        string[] currencies = ["USD", "EUR", "GBP", "JPY"];

        var listener = new HttpListener();
        listener.Prefixes.Add($"http://+:{port}/");
        listener.Start();
        Console.WriteLine($"http-order-processor starting  port={port}");

        while (true)
        {
            var ctx = await listener.GetContextAsync();
            if (ctx.Request.Url?.AbsolutePath == "/health")
            {
                ctx.Response.StatusCode = 200;
                await ctx.Response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes("{\"status\":\"ok\"}"));
                ctx.Response.Close(); continue;
            }
            if (ctx.Request.HttpMethod != "POST" || ctx.Request.Url?.AbsolutePath != "/intent")
            {
                ctx.Response.StatusCode = 404; ctx.Response.Close(); continue;
            }
            using var reader = new StreamReader(ctx.Request.InputStream);
            var body = await reader.ReadToEndAsync();
            ctx.Response.StatusCode = 200;
            await ctx.Response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes("{\"ack\":true}"));
            ctx.Response.Close();

            var envelope = JsonNode.Parse(body)?.AsObject();
            var intentId = SseHelper.Str(envelope ?? new JsonObject(), "intent_id");
            if (!string.IsNullOrEmpty(intentId))
            {
                _ = Task.Run(async () =>
                {
                    try
                    {
                        var intent = await client.GetIntentAsync(intentId);
                        var p = intent["payload"]?.AsObject() ?? new JsonObject();
                        var orderId = SseHelper.Str(p, "order_id");
                        var currency = SseHelper.Str(p, "currency");
                        JsonObject res;
                        if (SseHelper.Str(p, "event_type") != "order.placed") res = new JsonObject { ["action"] = "fail", ["reason"] = "unexpected event_type" };
                        else if (string.IsNullOrEmpty(orderId)) res = new JsonObject { ["action"] = "fail", ["reason"] = "order_id required" };
                        else if (!currencies.Contains(currency)) res = new JsonObject { ["action"] = "fail", ["reason"] = "unsupported currency" };
                        else res = new JsonObject { ["action"] = "complete", ["order_id"] = orderId, ["tracking_id"] = $"TRK-{orderId.ToUpper()}-{Math.Abs(orderId.GetHashCode()) % 100000:D5}", ["status"] = "accepted" };
                        await client.ResumeIntentAsync(intentId, res, new RequestOptions { OwnerAgent = addr });
                        Console.WriteLine($"resumed {intentId}");
                    }
                    catch (Exception ex) { Console.Error.WriteLine($"error: {ex.Message}"); }
                });
            }
        }
    }
}
